spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/feign/ribbon/FeignLoadBalancer.java

```diff
@@ -74,26 +95,32 @@ public RibbonResponse execute(RibbonRequest request, IClientConfig configOverrid
 		else {
 			options = new Request.Options(this.connectTimeout, this.readTimeout);
 		}
-		Response response = request.client().execute(request.toRequest(), options);
-		return new RibbonResponse(request.getUri(), response);
+		LoadBalancedRetryPolicy retryPolicy = loadBalancedRetryPolicyFactory.create(this.getClientName(), this);
+		retryTemplate.setRetryPolicy(retryPolicy == null ? new NeverRetryPolicy()
```

Comment:
I see, the retry policy created below uses the request. Could there be a problem since the retry policy is scoped to the request and the template is a field?